builtin/k8s/platform.go

```diff
@@ -153,7 +153,7 @@ func (p *Platform) Deploy(
 	// Build our env vars
 	env := []corev1.EnvVar{
 		{
-			Name:  "PORT",
+			Name:  "WAYPOINT_EXPOSED_PORT",
```

Comment:
I mean, I'm just changing it because I don't feel right abusing an environment variable like that - what is it used for?

Some applications might behave differently based on that `env` variable, and it woul be kind of unexpected for the user if their static_env is empty

Reply:
We have to leave this as `"PORT"` here for a couple of reasons:

Buildpacks expect this env var to exist as `"PORT"`. Also the URL service Waypoint uses is expecting `"PORT"` to exist in the environment. Changing the name will end up stalling or breaking the deploy all together for K8s. I'm guessing that's the main reason why the integration tests are failing on this PR too.

Reply:
Makes sense! Let's revert this change then 👍